src/lightning/fabric/utilities/load.py

```diff
@@ -203,3 +203,21 @@ def _load_tensor(t: _NotYetLoadedTensor) -> Tensor:
         return t._load_tensor()
 
     return apply_to_collection(collection, dtype=_NotYetLoadedTensor, function=_load_tensor)
+
+
+def _take_state_and_load_stateful(
+    source: Dict[str, Any], destination: Dict[str, Union[Any, _Stateful]], keys: Optional[Set[str]] = None
+) -> None:
+    """Takes the state from the source destination and moves it into the destination dictionary.
+
+    If an object in the destination follows the stateful protocol, it loads the source state via ``load_state_dict``.
+
+    """
+    keys = set(source.keys()) if keys is None else keys
```

Comment:
This is the same
```suggestion
    keys = set(source) if keys is None else keys
```

Or accessing the keys should work too

```suggestion
    keys = source.keys() if keys is None else keys
```